### PR TITLE
feat: add postgres-exporter

### PR DESCRIPTION
**Pull request by dspo**
#### What this PR does / why we need it:

安装网关 PostgreSQL 时, 一并安装 postgres_exporter

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | Install postgres_exporter along with the gateway PostgreSQL installation |
| 🇨🇳 中文    | 安装网关 PostgreSQL 时, 一并安装 postgres_exporter |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dspo** [comment]
@iutx @sfwn

File: internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go [hunk 1]
```diff
@@ -68,6 +68,9 @@ func (p *provider) BuildServiceGroupRequest(resourceInfo *handlers.ResourceInfo,
 	delete(req.GroupLabels, "LOCATION-CLUSTER-SERVICE")
 
 	for name, service := range resourceInfo.Dice.Services {
+		if name == "postgres-exporter" {
+			continue
+		}
 		delete(service.Envs, "LOCATION-CLUSTER-SERVICE")
 
 		// envs
```
Comment (dspo) [inline internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go:76]: postgre's dice.yaml
https://github.com/erda-project/erda-addons-enterprise/blob/master/addons/postgresql/1.0.0/dice.yml

File: internal/apps/msp/resource/deploy/handlers/apigateway/apigateway.go [hunk 2]
```diff
@@ -190,6 +189,22 @@ func (p *provider) BuildServiceGroupRequest(resourceInfo *handlers.ResourceInfo,
...
+				"spot-metrics",
+				"sign-auth",
+				"key-auth",
+				"rate-limiting",
+				"csrf-token",
+				"zone",
+				"host-check",
+				"hmac-auth",
+				"prometheus",
+			}, ","),
```
Comment (dspo) [inline internal/apps/msp/resource/deploy/handlers/apigateway/apigateway.go:207]: Kong 的默认插件写在环境变量中, 硬编码到代码里, 是否合适, 还是应该写到 kong addon 的  dice.yml 里?

File: internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go [hunk 3]
```diff
@@ -68,6 +68,9 @@ func (p *provider) BuildServiceGroupRequest(resourceInfo *handlers.ResourceInfo,
 	delete(req.GroupLabels, "LOCATION-CLUSTER-SERVICE")
 
 	for name, service := range resourceInfo.Dice.Services {
+		if name == "postgres-exporter" {
+			continue
+		}
```
Comment (dspo) [inline internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go:73]: postgre's dice.yaml
https://github.com/erda-project/erda-addons-enterprise/blob/master/addons/postgresql/1.0.0/dice.yml

File: internal/apps/msp/resource/deploy/handlers/apigateway/apigateway.go (hunk 2 above)
Comment (iutx) [reply internal/apps/msp/resource/deploy/handlers/apigateway/apigateway.go:207]: 可以看看后续是否变动频繁，当然可以写到 kong 的 dice.yml 中更合适

Comment (dspo) [reply internal/apps/msp/resource/deploy/handlers/apigateway/apigateway.go]: 已提交到 kong 的 dice.yaml

https://github.com/erda-project/erda-addons-enterprise/pull/56/files#diff-99a93e5bf2ff0e98c21183c208f21f2f443a8aae81b228654301bfeef7f094a2

File: build/dockerfiles/Dockerfile [hunk 4]
```diff
@@ -45,8 +43,10 @@ RUN \
     npm i -g jackson-converter@1.0.10 && \
     pip3 install dicttoxml xmindparser
 
-RUN wget  https://terminus-dice.oss-cn-hangzhou.aliyuncs.com/installer/nginx_1.20.2-1~bullseye_${ARCH}.deb && \
-        dpkg -i nginx_1.20.2-1~bullseye_${ARCH}.deb
+RUN arch="$(dpkg --print-architecture)"; arch="${arch##*-}"; \
```
Comment (sfwn) [inline build/dockerfiles/Dockerfile:46]: You just made some changes here, so let's make a judgment: only install niginx for hepa or all-in-one.

File: internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go (hunk 3 above)
Comment (sfwn) [reply internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go:73]: why ignored

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go (hunk 3 above)
Comment (dspo) [reply internal/apps/msp/resource/deploy/handlers/postgresql/postgresql.go:73]: 这个 for 循环后面的逻辑是对每一个 service 加环境变量、健康检查等，而这些已经在 addon 的 dice.yml 配置好了，无须在代码中再为其配置。

**Pull request closed without merging**